javatests/jsinterop/generator/externs/simpleclass/SimpleClass.java

```diff
@@ -50,6 +58,17 @@ public SimpleClass() {}
 
   public SimpleClass(String foo) {}
 
+  @Deprecated
+  public native boolean barMethod(String bar, String foo, JsObject baz);
+
+  @JsOverlay
+  public final boolean barMethod(String bar, String foo, Object baz) {
```

Comment:
This Overlay method should be deprecated as well. You need to fix java/jsinterop/generator/helper/ModelHelper.java:224 not to clear the Deprecated Annotation

Reply:
done